app/javascript/components/miq-toolbar.jsx

```diff
@@ -45,6 +45,18 @@ const getParams = (urlParms, sendChecked) => {
   return _.filter(params).join('&') || undefined;
 };
 
+const convertMultParamsToArray = (paramName, fullParamString) => {
+  let finalString = '';
+  let index = fullParamString.search(paramName);
+  while (index !== -1) {
+    finalString = `${finalString + fullParamString.slice(0, index + paramName.length)}[]`;
+    fullParamString = fullParamString.slice(index + paramName.length, fullParamString.length);
+    index = fullParamString.search(paramName);
+  }
+  finalString = finalString += fullParamString;
+  return finalString;
+};
+
```

Comment:
I'm not sure I like the manual parsing of the params string.  Is there a library that can do this for us?  If not, is there a way to write specs for this particular function?

Reply:
I figured out a way to make the filtering smarter so we avoid the issue you pointed out [here](https://github.com/ManageIQ/manageiq-ui-classic/pull/7861/files#r708513781) but it is still a manual parsing. I looked around for a bit but haven't yet found a library that does this sort of filtering.

Reply:
I pushed up the smarter (but still manual filtering) [here](https://github.com/ManageIQ/manageiq-ui-classic/pull/7861/files#diff-c893d0342b7a9c086d00c5cc07fe856d4af82bb0ed6435a3e5d579420ae11776R60-R87) in case you wanna look at it.